Kyle Kauzlarich

kyle-kauzlarich opened the pull request:
all commands working besides cd 
questions answered
Single Platform shell requires OSX to run 
tested on high sierra

kyle-kauzlarich committed:
Questions Added
and answers written out

kyle-kauzlarich committed:
Additions
i can call any of the comands that are stored in /bin/ and get a 'usage' responce next step is to bake in the cd command and add compatability to commands that take more than just typing in the command

kyle-kauzlarich committed:
slight changes
i have the single argument commands working like ls and pwd
next plan of action is to get the 2 argument comands working like touch and the like then get the cd command made

kyle-kauzlarich committed:
Changes
changed a line to get the commands that take 2 inputs working

kyle-kauzlarich committed:
formatted

cwindeknecht commented:
## Commit Guideline
### I have generalized these as opposed to giving you personalized commit critiques.  Surely you can discern for yourselves which of these recommendations you have or have not been following.  
#### If you are already following all of them, good on ya.  If you aren't, for your sakes to be prepared in the professional world / be able to pinpoint where in code you need to look to see what changes you made more easily, please begin to do so.
- Commit constantly!  Don't go overboard and do it with every line of code, but any time you finish a component, write an essential function, etc. just throw a commit up there. 
- Make sure to use a title and message.  If you are doing this from the command line, it will look like this: -m "Title" -m " " -m "Message".
- Think of what the title should be similar to if you were writing a book.  "part 1 done" doesn't tell you much unless you take the time to go find out what part 1 entailed.  Instead "Function for Dynamic File Reading Completed" is pretty obvious what you did.
- While messages aren't always necessary, it is still usually a good idea to include them anyways.  From the above example, an ideal message would be "filename.js --- used readFile and regex to parse the program correctly."  Is that necessary?  Not really.  Later on, if you forgot where you were doing the dynamic file loading and you had that message, you wouldn't have to dig through the files to find where it was.  That doesn't really apply to our tiny programs that have a few files, but if you have way more it can really help. Think back to the front-end project for instance.  
- In general, you should only commit working code.  If you like to self-document what you did wrong to better your understanding, then a broken commit here and there isn't the end of the world.  I would suggest though, that if you want to do this, title it something akin to "Unfinished Dynamic File Reading" or "Broken Dynamic File Reading".  


## Code Review
* MVP achieved.  Checked for all the necessary errors.  Every command works.  Code very uniquely, so much so that I can't really make recommendations unfortunately on how to make it better.  
* The only thing I noticed was that a continue statement was necessary in the instance of a change directory in order to keep it from throwing an error, but it still works anyways.

kyle-kauzlarich committed:
cleaned up
debugging output logs commited out
re introduced the if statement to stop from exit and cd
changed prompt name to Vanilla-Shell
